### ISSUE TITLE
configure failed on Windows7 Chinese version

### ISSUE THREAD

**Issue by jhenchen** [objectcomputing/OpenDDS#1110]
I'm using My VS2015 command line tool with the configure file.But It failed.The complete command line output is :
![output](https://user-images.githubusercontent.com/18344904/56998656-a3692700-6bde-11e9-9975-0723ce1030e1.png)
I have read the issue #762, but It maybe a new issue.

**Comment by mitza-oci** [comment]
At least for the short term, you can hard-code the version in your local copy of the configure script.  

Perhaps the best way to support it would be to accept a command line argument for configure like `--vs-version=2015`.

**Comment by jhenchen** [comment]
Thanks.

**Comment by tiedun001** [comment]
change IDE to VS2017 and download the newest opendds version and others,then ./configure to make.

**Comment by iguessthislldo** [comment]
I'm wondering if [vswhere](https://github.com/Microsoft/vswhere) could help with this for newer versions of VS that have it.

I took of the batch examples and modified it slightly:
```
@echo off
setlocal enabledelayedexpansion

set PATH=%PATH%;%ProgramFiles(x86)%\Microsoft Visual Studio\Installer

for /f "usebackq tokens=*" %%i in (`vswhere -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath`) do (
  set InstallDir=%%i
)

if exist "%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt" (
  set /p Version=<"%InstallDir%\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt"

  rem Trim
  set Version=!Version: =!
)

echo %Version%
pause
```
On my work laptop with VS2019 this results in `14.20.27508`.

It's not the exact same as the version of `cl` but MSBuild's version is always 5 major versions less than it's `cl`. Of course maybe if we're already in the vs dev environment, we can bypass vswhere and use the file with the version directly.

**Comment by mitza-oci** [comment]
@iguessthislldo I don't want to have to maintain a complex script with dependencies on the locations of things on the filesystem and the compiler target architecture.  It would be easier to run `cl` and ask it to dump `_MSC_VER`.

**Comment by mitza-oci** [comment]
> change IDE to VS2017 and download the newest opendds version and others,then ./configure to make.

I don't understand what this comment means.  Does the upgrade from VS 2015 to 2017 resolve the issue?

**Comment by mitza-oci** [comment]
Here's the approach I was thinking of a few weeks ago.  It's independent of localization and doesn't require extra dependencies.

1. In the configure script, open a temporary file using `File::Temp`.  
2. Write to it the following using a here-document:
```
#ifdef _M_X64
#define ARCH x64
#else
#define ARCH x86
#endif
_MSC_VER ARCH
```
3. Change the part of the configure script that runs cl to run `cl /EP $tempfile` and pipe only its stdout (not stderr) to a Perl filehandle
4. Update the parsing code, the only nonblank line will look like:
```
1920 x64
```

**Comment by iguessthislldo** [comment]
This should be fixed by #1141
